test/e2e/service.go

```diff
@@ -1152,6 +1152,53 @@ var _ = framework.KubeDescribe("Services", func() {
 			jig.TestHTTPHealthCheckNodePort(publicIP, healthCheckNodePort, "/healthz", expectedSuccess)
 		}
 	})
+
+	It("should only allow access from service loadbalancer source ranges [Slow]", func() {
+		// this feature currently supported only on GCE/GKE/AWS
+		framework.SkipUnlessProviderIs("gce", "gke", "aws")
+
+		loadBalancerCreateTimeout := loadBalancerCreateTimeoutDefault
+		largeClusterMinNodesNumber := 100
+		if nodes := framework.GetReadySchedulableNodesOrDie(c); len(nodes.Items) > largeClusterMinNodesNumber {
+			loadBalancerCreateTimeout = loadBalancerCreateTimeoutLarge
+		}
+
+		namespace := f.Namespace.Name
+		serviceName := "lb-sourcerange"
+		jig := NewServiceTestJig(c, serviceName)
+
+		By("Prepare allow source ips")
+		// prepare the exec pods
+		// hostPod and acceptPod are allowed to access the loadbalancer
+		hostPod := framework.LaunchHostExecPod(c, namespace, "hostexec-accept")
+		acceptPodName := createExecPodOrFail(f.Client, namespace, "execpod-accept")
+		dropPodName := createExecPodOrFail(f.Client, namespace, "execpod-drop")
+
+		hostPod, err := c.Pods(namespace).Get(hostPod.Name)
+		Expect(err).NotTo(HaveOccurred())
+		accpetPod, err := c.Pods(namespace).Get(acceptPodName)
+		Expect(err).NotTo(HaveOccurred())
+
+		By("creating a pod to be part of the service " + serviceName)
+		// This container is an nginx container listening on port 80
+		// See kubernetes/contrib/ingress/echoheaders/nginx.conf for content of response
+		jig.RunOrFail(namespace, nil)
+		// Create loadbalancer service with source range from node[0] and podAccept
+		svc := jig.CreateTCPServiceOrFail(namespace, func(svc *api.Service) {
+			svc.Spec.Type = api.ServiceTypeLoadBalancer
+			svc.Spec.LoadBalancerSourceRanges = []string{hostPod.Status.HostIP + "/32", accpetPod.Status.PodIP + "/32"}
```

Comment:
It was not specified in the doc though. 
https://cloud.google.com/compute/docs/networking#addingafirewall